lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1442,19 +1445,27 @@ class HSAQueue final : public KalmarQueue
 
 
 
-        if (!drainingQueue_ && (asyncOps.size() >= MAX_INFLIGHT_COMMANDS_PER_QUEUE-1)) {
+        if (!drainingQueueFailed_ && !drainingQueue_ && (asyncOps.size() >= HCC_MAX_INFLIGHT_COMMANDS_PER_QUEUE-1)) {
```

Comment:
a second re-read of the logic here makes it hard to distinguish which value is variable, and which is constant.

to my understanding, originally `HCC_MAX_INFLIGHT_COMMANDS_PER_QUEUE` was a constant (macro), but now it's a global variable modifiable by environment variables. it's less than desirable.

Reply:
.. but I think we can live with it for now as long as these values are set / accessed in a consistent fashion.